src/main/java/dk/kb/discover/webservice/OpenApiResource.java

```diff
@@ -0,0 +1,148 @@
+package dk.kb.discover.webservice;
+
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import dk.kb.discover.config.ServiceConfig;
+import dk.kb.discover.util.DsDiscoverClient;
+import dk.kb.util.Resolver;
+import dk.kb.util.webservice.exception.InvalidArgumentServiceException;
+import dk.kb.util.yaml.YAML;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.error.YAMLException;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class OpenApiResource {
+    private static final Logger log = LoggerFactory.getLogger(OpenApiResource.class);
+
+
+    public static final String APPLICATION_YAML = "application/yaml";
+
+    /**
+     * Pattern to allow search-replace for variabels defined as ${config.yaml.path} in OpenAPI specifications.
+     * Everything after 'config.' is treated as a path to an entry in the backing configuration.
+     */
+    private static final Pattern CONFIG_REPLACEMENT= Pattern.compile("\\$\\{config:([^}]+)\\}");
+
+    /**
+     * Deliver the OpenAPI specification with substituted configuration values as a YAML file.
+     */
+    @GET
+    @Produces(APPLICATION_YAML)
+    @Path("/{path}.yaml")
+    public Response getYamlSpec(@PathParam("path") String path) {
+        try {
+            String inputYaml = Resolver.readFileFromClasspath(path + ".yaml");
+
+            if (inputYaml == null){
+                throw new InvalidArgumentServiceException("No OpenAPI specification with path '" + path + ".yaml' was found.");
```

Comment:
Isn't this a plain 'ol 404? (`NotFoundServiceException`)

Reply:
Yeah, good point